lib/srv/discovery/database_watcher.go

```diff
@@ -124,11 +125,14 @@ func (s *Server) onDatabaseDelete(ctx context.Context, resource types.ResourceWi
 	return trace.Wrap(s.AccessPoint.DeleteDatabase(ctx, database.GetName()))
 }
 
-func filterResourcesByOrigin[T types.ResourceWithOrigin, S ~[]T](all S, wantOrigin string) (filtered S) {
+func filterResourcesByOrigin[T types.ResourceWithLabels, S ~[]T](all S, wantOrigin, wantResourceGroup string) (filtered S) {
 	for _, resource := range all {
-		if resource.Origin() == wantOrigin {
-			filtered = append(filtered, resource)
+		resourceDiscoveryGroup, _ := resource.GetLabel(types.TeleportInternalDiscoveryGroupName)
+		if resource.Origin() != wantOrigin || resourceDiscoveryGroup != wantResourceGroup {
```

Comment:
Does this mean the default value of the group is empty, and it will match resources prior to this change for backward compat? Just want to confirm

Reply:
If users don't define the new parameter, yes. 
I also added an extra validation that ensures the discovery group is updated if the resource already exists in auth server when `onCreate` is invoked because the discovery group does not match (migrating from previous version or changing the discovery group value)